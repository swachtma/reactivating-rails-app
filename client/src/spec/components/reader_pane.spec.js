/* global expect, jest */
import React from 'react'
import { shallow } from 'enzyme'

import configEnzyme, { expectXInY } from '../../test_helper'
import ReaderPane from '../../components/reader_pane'

let nodes = ['#reader-pane', 'Connect(BookmarkModal)', 'Connect(NodesList)', 'Connect(ChapterFooter)']

describe('<ReaderPane />', () => {
  let w = shallow(<ReaderPane />)
  nodes.forEach((node) => {
    it('renders ' + node, () => { expectXInY(node, w) })
  })

  it('renders according to snapshot', () => expect(w).toMatchSnapshot())
})
