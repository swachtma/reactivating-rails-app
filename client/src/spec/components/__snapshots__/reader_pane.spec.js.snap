// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReaderPane /> renders according to snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReaderPane />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(BookmarkModal) />,
        <Container
          text={true}
        >
          <Connect(NodesList) />
          <Connect(ChapterFooter) />
        </Container>,
      ],
      "id": "reader-pane",
      "style": Object {
        "overflowWrap": "break-word",
        "paddingBottom": "15px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Connect(NodesList) />,
            <Connect(ChapterFooter) />,
          ],
          "text": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(BookmarkModal) />,
          <Container
            text={true}
          >
            <Connect(NodesList) />
            <Connect(ChapterFooter) />
          </Container>,
        ],
        "id": "reader-pane",
        "style": Object {
          "overflowWrap": "break-word",
          "paddingBottom": "15px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Connect(NodesList) />,
              <Connect(ChapterFooter) />,
            ],
            "text": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
